content/realtime/channels/channel-parameters/overview.textile

```diff
@@ -0,0 +1,82 @@
+---
+title: Channel Parameters
+section: realtime
+index: 1
+jump_to:
+  Help with:
+    - Overview#overview
+    - Supported channel parameters#supported-parameters
+    - Using channel parameters#using-parameters
+    - Using channel parameters with non-Ably transports#using-parameters-with-other-transports
+    - Next steps#next-steps
+---
+
+h2(#overview). Overview
+
+Channel parameters allow a client to set the properties of a channel. A set of channel parameters is a set of key/value pairs, where both keys and values are strings; the keys correspond to specific features that Ably defines.
+
+The methods provided for specifying channel parameters are outlined below.
+
+h2(#supported-parameters). Supported channel parameters
+
+- rewind := Used to request that an attachment start from a given number of messages or point in time in the past. See "rewind":/realtime/channels/channel-parameters/rewind for more information.
+- delta := Used to request that data payloads should be sent as deltas to the previous payload. See "deltas":/realtime/channels/channel-parameters/deltas for more information.
+
+h2(#using-parameters). Using channel parameters
+
+You can specify channel parameters in the @ChannelOptions@ when obtaining a @Channel@. A collection of channel parameters is expressed as a map of string key/value pairs. The @ChannelOptions@ associated with a channel may also be updated by calling "setOptions":/realtime/channels/overview#modifying-options. The parameters associated with a channel take effect when the channel is first attached; if the parameters are subsequently modified via a call to @setOptions@, then that call triggers an attach operation that applies the updated parameters, if successful.
+
+h3. Example
+
+For example, to specify the @rewind@ channel param with the value @"1"@:
+
+```[javascript]
+  const realtime = new Ably.Realtime('{{API_KEY}}');
+  const channelOpts = {params: {rewind: '1'}}
+  const channel = realtime.channels.get('{{RANDOM_CHANNEL_NAME}}', channelOpts);
+```
+
+To modify the @rewind@ channel param with the value @"15s"@:
+
+```[javascript]
+  const realtime = new Ably.Realtime('{{API_KEY}}');
+  const channelOpts = {params: {rewind: '15s'}}
+  channel.setOptions(channelOpts, (err) => {
+    if(!err) {
+      console.log('channel params updated')
+    }
+  });
+```
+
+h2(#using-parameters-with-other-transports). Using parameters with non-Ably transports
```

Comment:
Can't this also be used with SDKs that don't yet support channel params?

Reply:
I'm assuming not, @paddybyers?

Reply:
@paddybyers would you be happy for us to add this back in?

> h2(#using-params-with-lib-v11). Using channel params with v1.1 or earlier Ably client libraries
> The current Ably libraries, at version 1.1, do not expose an API for expressing channel parameters. This means that it is necessary to specify parameters in a way that is opaque to the library.
> A set of params is expressed by including a query string, using standard URL query syntax and encoding, within the qualifier part of a channel name. The qualifier part is in square brackets at the start of the channel name.
> For example, to specify the parameter @foo@ with value @bar@ on channel @baz@, the qualified channel name would be @[?foo=bar]baz@. If the channel name already has a qualifier, such as @[meta]log@, then the query string follows the existing qualifier, as in @[meta?foo=bar]log@.
> Using this syntax with the Ably library means that channel parameters are specified for the lifetime of the @Channel@ instance; in order to reference the same channel, but with different channel parameters, it is necessary to get a new @Channel@ instance, using a qualified name that includes the new channel parameters.
> h3. Example
> For example, to specify the @rewind@ channel param with the value @"1"@:
> ```[javascript]
>   const realtime = new Ably.Realtime('{{API_KEY}}');
>   const channel = realtime.channels.get('[?rewind=1]{{RANDOM_CHANNEL_NAME}}');
> ```

Reply:
yes

Reply:
updated in latest commit https://github.com/ably/docs/pull/880/commits/e82a6804ec69b6f985ca69d98d8a4979645392fa